test/performance/io/format_fasta_benchmark.cpp

```diff
@@ -0,0 +1,202 @@
+#if __has_include(<seqan/seq_io.h>)
+    #include <seqan/seq_io.h>
+#endif
+
+#include <algorithm>
+#include <cctype>
+#include <cstring>
+
+#include <benchmark/benchmark.h>
+
+#include <sstream>
+
+#include <range/v3/algorithm/equal.hpp>
+#include <range/v3/view/transform.hpp>
+
+#include <seqan3/alphabet/quality/all.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/io/sequence_file/input_format_concept.hpp>
+#include <seqan3/io/sequence_file/output.hpp>
+#include <seqan3/io/sequence_file/output_format_concept.hpp>
+#include <seqan3/io/sequence_file/format_fasta.hpp>
+#include <seqan3/range/view/convert.hpp>
+
+using namespace seqan3;
+using namespace seqan3::literal;
+
+static void write3(benchmark::State& state)
+{
+    std::stringstream ostream; //{"> seq\nACTAGACTAGCTACGATCAGCTACGATCAGCTACGA\n"};
+    sequence_file_format_fasta format;
+    sequence_file_output_options options;
+    std::string id{"seq"};
+    dna5_vector seq{"ACGTAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"_dna5};
+
+    for (auto _ : state)
+    {
+        format.write(ostream, options, seq, id, std::ignore);
+    }
+}
+
+BENCHMARK(write3);
+
+#if __has_include(<seqan/seq_io.h>)
+
+    static void write2(benchmark::State& state)
+    {
+        std::stringstream outStream;
+        seqan::CharString meta = "seq";
+        seqan::Dna5String seq = "ACGTAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+
+        for (auto _ : state)
+        {
+            seqan::writeRecord(outStream, meta, seq, seqan::Fasta());
+        }
+    }
+
+    BENCHMARK(write2);
+#endif
+
+static void read3(benchmark::State& state)
+{
+    std::string dummy_file{};
+    for (size_t idx = 0; idx < 10000000; idx++)
+        dummy_file += ">seq\nACTAGACTAGCTACGATCAGCTACGATCAGCTACGA\n";
+    std::stringstream istream{dummy_file};
+    sequence_file_format_fasta format;
+    sequence_file_input_options<dna5, false> options;
+    std::string id;
+    dna5_vector seq;
+    for (auto _ : state)
+    {
+        format.read(istream, options, seq, id, std::ignore);
+    }
+}
+BENCHMARK(read3);
+
+static void read3_from_file(benchmark::State& state)
+{
+    // Create fasta file
+    std::string filename{"Test_Format_Fasta_Benchmark.fasta"};
+    sequence_file_output fout{filename};
+    for (size_t idx = 0; idx < 10000000; idx++)
+    {
+        std::string id{ "seq" };
+        dna5_vector seq{ "ACTAGACTAGCTACGATCAGCTACGATCAGCTACGA"_dna5 };
+        fout.push_back(std::tie(seq, id));   // as a tuple
+    }
+
+    sequence_file_input fin{filename};
+    auto file_it = fin.begin();
+    std::string id;
+    dna5_vector seq;
+
+    for (auto _ : state)
+    {
+        auto record = *file_it;
+        ++file_it;
+    }
+}
+
+BENCHMARK(read3_from_file);
+
+
+#if __has_include(<seqan/seq_io.h>)
+
+#include <fstream>
+
+std::fstream* createFastAFile(seqan::CharString &tempFilename)
+{
+    using namespace seqan;
+
+    tempFilename = SEQAN_TEMP_FILENAME();
+    char filenameBuffer[1000];
```

Comment:
I think your code for writing a file is a little complicated. You can just do this:
```cpp
  std::ofstream myfile;
  myfile.open (toCString(tempFilename));
  SEQAN_ASSERT(file->is_open());

  for (size_t idx = 0; idx < 10000000; idx++)
      myfile << ">seq\nACTAGACTAGCTACGATCAGCTACGATCAGCTACGA\n";
  
  myfile.close();
```

Reply:
Okay, but that code is from a Seqan2 test file. Do you want it changed there as well?